dss-apps/dss-datamodel-center/dss-datamodel-dimension-management/src/main/java/com/webank/wedatasphere/dss/datamodel/dimension/service/impl/DimensionServiceImpl.java

```diff
@@ -0,0 +1,214 @@
+        enableOne.setUpdateTime(new Date());
+        return getBaseMapper().update(enableOne, Wrappers.<DssDatamodelDimension>lambdaUpdate().eq(DssDatamodelDimension::getId,id));
+    }
+
+
+    @Override
+    @Transactional(rollbackFor = Exception.class)
+    public int updateDimension(Long id, DimensionUpdateVO vo) throws DSSDatamodelCenterException{
+        DssDatamodelDimension org = getBaseMapper().selectById(id);
+        if (org == null) {
+            LOGGER.error("errorCode : {}, update dimension error not exists", ErrorCode.DIMENSION_UPDATE_ERROR.getCode());
+            throw new DSSDatamodelCenterException(ErrorCode.DIMENSION_UPDATE_ERROR.getCode(), "update dimension error not exists");
+        }
+
+        if (datamodelReferencService.dimensionReferenceCount(org.getName())>0
+                ||datamodelReferencService.dimensionReferenceCount(org.getFieldIdentifier())>0
+                ||datamodelReferencService.dimensionReferenceCount(vo.getName())>0
+                ||datamodelReferencService.dimensionReferenceCount(vo.getFieldIdentifier())>0){
+            LOGGER.error("errorCode : {}, dimension name can not  referenced", ErrorCode.DIMENSION_UPDATE_ERROR.getCode());
+            throw new DSSDatamodelCenterException(ErrorCode.DIMENSION_UPDATE_ERROR.getCode(), "dimension name can not  referenced");
+        }
+
+
+        //当更新名称时
+        if (!StringUtils.equals(vo.getName(), org.getName())) {
+            int repeat = getBaseMapper().selectCount(Wrappers.<DssDatamodelDimension>lambdaQuery().eq(DssDatamodelDimension::getName, vo.getName()));
+            if (repeat > 0) {
+                LOGGER.error("errorCode : {}, dimension name can not repeat ", ErrorCode.DIMENSION_UPDATE_ERROR.getCode());
+                throw new DSSDatamodelCenterException(ErrorCode.DIMENSION_UPDATE_ERROR.getCode(), "dimension name can not repeat ");
+            }
+        }
+
+        String orgFieldIdentifier = org.getFieldIdentifier();
+        //当更新标识时
+        if (!StringUtils.equals(vo.getFieldIdentifier(), orgFieldIdentifier)) {
+            int repeat = getBaseMapper().selectCount(Wrappers.<DssDatamodelDimension>lambdaQuery().eq(DssDatamodelDimension::getFieldIdentifier, vo.getFieldIdentifier()));
+            if (repeat > 0) {
+                LOGGER.error("errorCode : {}, dimension field identifier can not repeat", ErrorCode.DIMENSION_UPDATE_ERROR.getCode());
+                throw new DSSDatamodelCenterException(ErrorCode.DIMENSION_UPDATE_ERROR.getCode(), "dimension field identifier can not repeat ");
+            }
+        }
+        DssDatamodelDimension updateOne = modelMapper.map(vo,DssDatamodelDimension.class);
+        updateOne.setUpdateTime(new Date());
+        getBaseMapper().update(updateOne, Wrappers.<DssDatamodelDimension>lambdaUpdate().eq(DssDatamodelDimension::getId,id));
+
+        //同步atlas
+        UpdateModelTypeResult updateModelTypeResult = assertsSyncService.syncUpdateModel(
```

Comment:
publisher.publishEvent(new UpdateModelEvent(this
                ,DataModelSecurityContextHolder.getContext().getDataModelAuthentication().getUser()
                ,vo.getFieldIdentifier()
                ,orgFieldIdentifier
                ,ClassificationConstant.DIMENSION));